src/index.js

```diff
@@ -0,0 +1,89 @@
+// import './style.css';
+
+const test = /[0-9]/g;
+const _ = require('lodash');
+const { printWeather } = require('./components/printComponents');
+const { setLocal, printHistory } = require('./components/history');
+
+function HistoryObject(data) {
+  this.location = `${data.location.country}, ${data.location.region}`;
+  this.temperature = data.current.temperature;
+  this.windDir = data.current.wind_dir;
+  this.windSpeed = data.current.wind_speed;
+  const icon = data.current.weather_icons[0];
+  this.icon = icon;
+  this.time = data.location.localtime.slice(11, 16);
+  this.pressure = data.current.pressure;
+  this.feelslike = data.current.feelslike;
+  this.discription = data.current.weather_descriptions;
+}
+
+const fetchAPI = (coord) => {
+  const city = document.querySelector('.header__input--city').value || coord;
+  if (coord || !city.match(test)) {
+    fetch(
+      `http://api.weatherstack.com/current?access_key=bdeec7a4beb014daa4415d15b0447360&query=${city}`
+    )
+      .then((data) => data.json())
+      .then((data) => {
+        const town = new HistoryObject(data);
+        printWeather(town);
+        setLocal(town);
+      });
+  }
+};
```

Comment:
```suggestion
const fetchAPI = (coord) => {
  const city = document.querySelector('.header__input--city').value;
  if (coord || !city.match(test)) {
    fetch(
      `http://api.weatherstack.com/current?access_key=bdeec7a4beb014daa4415d15b0447360&query=${city || coord}`
    )
      .then((data) => data.json())
      .then((data) => {
        const town = getHistoryObject(data);
        printWeather(town);
        setLocal(town);
      });
  }
};
```
С учетом переписанной функции выше можно переписать эту таким образом